code/modules/mob/dead/new_player/poll.dm

```diff
@@ -513,14 +517,21 @@
 	if(!replytext)
 		to_chat(usr, "The text you entered was blank. Please correct the text and submit again.")
 		return
-	var/adminrank = sanitizeSQL(poll_check_voted(pollid, TRUE))
+	var/voted = poll_check_voted(pollid, TRUE)
+	if(isnull(voted))
+		return
+	var/adminrank = sanitizeSQL(poll_rank())

```

Comment:
The procs for each poll type are split between this proc and doing it inline, could you standardise the one or the other?

Reply:
Actually all the procs are a bit of a hodgepode, bleh.

Reply:
I'll take a look.